packages/components/src/Interactive.ts

```diff
@@ -35,124 +35,133 @@ const allowedHoverVariables = [
   'boxShadow',
   'opacity',
   'textDecoration',
-];
+] as const;
+
+// Union type of all values in allowedHoverVariables.
+type HoverStyleProperty = (typeof allowedHoverVariables)[number];
+
 const pointerTags = ['button', 'area', 'a', 'select'];
 const pointerTypes = ['radio', 'checkbox', 'submit', 'button'];
 
-export type InteractiveProps = {
-  children?: ReactNode;
-  className?: string;
-  disabled?: boolean;
-  hover?: React.CSSProperties;
-  role?: string;
-  style?: React.CSSProperties;
-  tabIndex?: number;
+// Since Interactive may render any HTML element based on the tagName prop,
+// use AllHTMLAttributes<HTMLElement> to allow this to accept attributes
+// for any HTML element.
+// TODO: Can a more specific type be used by parameterizing this with a tag name?
+export interface InteractiveProps extends AllHTMLAttributes<HTMLElement> {
+  /** Styles applied when element is hovered or focused. */
+  hover?: Pick<React.CSSProperties, HoverStyleProperty>;
+
+  /** HTML tag to render. */
   tagName?: keyof JSX.IntrinsicElements;
-  type?: string;
-  onBlur?: (event: any) => void;
-  onClick?: (event: any) => void;
-  onKeyDown?: (event: any) => void;
-  onMouseDown?: (event: any) => void;
-};
-
-export const Interactive: React.ForwardRefExoticComponent<InteractiveProps> = forwardRef(
-  (
+
+  // Allow arbitrary data attributes on the rendered element.
+  [dataAttribute: `data-${string}`]: string;
```

Comment:
Since `Interactive` largely passes through props to the rendered HTML element, this type allows Interactive to accept arbitrary `data-*` props (similar to how react-hyperscript-helpers for HTML elements like `div`, `button`, etc. do). JS code that has a reference to the HTML element could access the values of these attributes via the `dataset` property.